client/src/portal/mod.rs

```diff
@@ -304,6 +304,19 @@ impl Keyring {
         }
         self.key.get().unwrap()
     }
+
+    /// Change keyring secret
+    ///
+    /// # Arguments
+    ///
+    /// * `secret` - The new secret to store.
+    pub async fn change_secret(&mut self, secret: Secret) -> Result<(), Error> {
```

Comment:
I wonder if this should accept the more general `&[u8]` and use Secret as an implementation detail inside.

Reply:
Yeah it make sense. But, I kind of like to follow the `Keyring::load()` approach :)